tools/base/fpgainfo/board.c

```diff
@@ -164,7 +165,7 @@ int unload_board_plugin(void)
 			if (res) {
 				char *err = dlerror();
 				OPAE_ERR("dlclose failed with %d %s", res, err ? err : "");
-				res = FPGA_EXCEPTION;
+				resval = FPGA_EXCEPTION;

```

Comment:
Let's pretend you have 10 plugins. This means the for-loop will iterate 10 times and call dlclose each time.
Now let's say  that dlclose  has an error at iteration N and returns non-zero. This means resval will get this value.
Next iteration, N+1, succeeds when calling dlclose. This successful call overwrites resval and we never know that there was an error closing.

Reply:
Unload called while closing fpgainfo application, it closes all loaded/ Open modules/ shared libs.
resval  is set to FPGA_EXCEPTION when unload files. value is stored  and returns end of this function.